### ISSUE TITLE
double escaping on nested json string

### ISSUE THREAD

**Issue by huanggze** [fluent/fluent-bit#1044]
## Bug Report

**Describe the bug**

Hi, fluentbit people. I am facing a double escaping problem which is sort of related to [#615](https://github.com/fluent/fluent-bit/issues/615#issue-329400149 )

**To Reproduce**
The only difference with #615 is in that the failed log contains nested json string, for example. my log is as follow
```
{"id":"3","log":{"level_1":"{\"apiVersion\":\"batch/v1beta1\",\"kind\":\"CronJob\"}"}}
```
**Expected behavior**

I would expect the output as (the log example mimicks k8s advanced auditing log):
```
{"id":"3","log":{"level_1":"{\"apiVersion\":\"batch/v1beta1\",\"kind\":\"CronJob\"}"}}
```
but the output is actually:
```
{"id":"3", "log":{"level_1":"{\\\"apiVersion\\\":\\\"batch/v1beta1\\\",\\\"kind\\\":\\\"CronJob\\\"}"}}
```

**Your Environment**

* Version used: 1.0.2
* Environment name and version: Kubernetes 1.10.5
* Filters and plugins:
my fluent-bit-config.yaml is as follows"
```
[SERVICE]
    Flush         1
    Log_Level     info
    Daemon        off
    Parsers_File  parsers.conf

[INPUT]
    Name              tail
    Tag               audit.*
    Path              /var/log/containers/sample.log
    Parser            docker
    DB                /var/log/flb_kube.db
    Mem_Buf_Limit     5MB
    Skip_Long_Lines   Off
    Refresh_Interval  10

[OUTPUT]
    Name file
    Match audit.*
    Path /var/log/output.txt

[PARSER]
    Name        docker
    Format      json
    # Command      |  Decoder | Field | Optional Action
    # =============|==================|=================
    Decode_Field_As   escaped    log
    Decode_Field_As   escaped    level_1
```

Note that I have no problem with `{"id":"2", "log":"hello \"AAA\" world"}` formmated log. And I can get a correct output as `{"id":"2", "log":"hello \"AAA\" world"}`

I am not sure if it is a bug or just because I miss some necessary config data

**Comment by edsiper** [comment]
Please check the following comment on #1278 : 

https://github.com/fluent/fluent-bit/issues/1278#issuecomment-499583503

**Comment by edsiper** [comment]
Issue already fixed, ref: #1278 (comment)
